packages/twenty-front/src/modules/settings/data-model/validation-schemas/settingsDataModelObjectAboutFormSchema.ts

```diff
@@ -0,0 +1,68 @@
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+import { t } from '@lingui/core/macro';
+import { ZodType, z } from 'zod';
+import { ReadonlyKeysArray } from '~/types/ReadonlyKeysArray';
+import { zodNonEmptyString } from '~/types/ZodNonEmptyString';
+import { camelCaseStringSchema } from '~/utils/validation-schemas/camelCaseStringSchema';
+
+type ZodTypeSettingsDataModelFormFields = ZodType<
+  Pick<
+    ObjectMetadataItem,
+    | 'labelSingular'
+    | 'labelPlural'
+    | 'description'
+    | 'icon'
+    | 'namePlural'
+    | 'nameSingular'
+    | 'isLabelSyncedWithName'
+  >
+>;
+const settingsDataModelFormFieldsSchema = z.object({
+  description: z.string().nullish(),
+  icon: z.string().optional(),
+  labelSingular: zodNonEmptyString,
+  labelPlural: zodNonEmptyString,
+  namePlural: zodNonEmptyString.and(camelCaseStringSchema),
+  nameSingular: zodNonEmptyString.and(camelCaseStringSchema),
+  isLabelSyncedWithName: z.boolean(),
+}) satisfies ZodTypeSettingsDataModelFormFields;
+
+export const settingsDataModelObjectAboutFormSchema =
+  settingsDataModelFormFieldsSchema.superRefine(
+    ({ labelPlural, labelSingular, namePlural, nameSingular }, ctx) => {
+      const labelsAreDifferent =
+        labelPlural.trim().toLowerCase() !== labelSingular.trim().toLowerCase();
+      if (!labelsAreDifferent) {
+        const labelFields: ReadonlyKeysArray<ObjectMetadataItem> = [
```

Comment:
For the record, why do you create this ReadonlyKeysArray type ? What do you wanna avoid here ?

Reply:
I don't wanna have stale untyped strings array below:
```ts
        [
           'labelPlural',
          'labelSingular',
        ];
```
The idea here is that by strict typing these readonly strings, on an `ObjectMetadataItem` `labelPlural` or `labelSingular` mutation `TypeScript`would raise an error